modules/ingest-service-impl/src/main/java/org/opencastproject/ingest/scanner/Ingestor.java

```diff
@@ -68,100 +80,218 @@
+    private final File artifact;
+    private int retryCount;
+    private boolean failed;
+    private RateLimiter throttle;
+
+    RetriableIngestJob(final File artifact, int secondsBetweenTries) {
+      this.artifact = artifact;
+      this.retryCount = 0;
+      this.failed = false;
+      throttle = RateLimiter.create(1.0 / secondsBetweenTries);
+    }
+
+    public boolean hasFailed() {
+      return this.failed;
+    }
+
+    public int getRetryCount() {
+      return this.retryCount;
+    }
+
+    public File getArtifact() {
+      return this.artifact;
+    }
+
+    @Override
+    public RetriableIngestJob call() {
+      return secCtx.runInContext(() -> {
+          if (hasFailed()) {
+            logger.warn("This is retry number {} for file {}. We will wait for {} seconds before trying again",
+                    retryCount, artifact.getName(), secondsBetweenTries);
+            throttle.acquire();
+          }
+          try (InputStream in = new FileInputStream(artifact)) {
+            failed = false;
+            ++retryCount;
+            if ("zip".equalsIgnoreCase(FilenameUtils.getExtension(artifact.getName()))) {
+              logger.info("Start ingest inbox file {} as a zipped mediapackage", artifact.getName());
+              WorkflowInstance workflowInstance = ingestService.addZippedMediaPackage(in, workflowDefinition, workflowConfig);
+              logger.info("Ingested {} as a zipped mediapackage from inbox as {}. Started workflow {}.",
+                      artifact.getName(), workflowInstance.getMediaPackage().getIdentifier().compact(),
+                      workflowInstance.getId());
+            } else {
+              /* Create MediaPackage and add Track */
+              MediaPackage mp = ingestService.createMediaPackage();
+              logger.info("Start ingest track from file {} to mediapackage {}",
+                      artifact.getName(), mp.getIdentifier().compact());
+
+              /* Check if we have a subdir and if its name matches an existing series */
+              File dir = artifact.getParentFile();
+              String seriesID = "";
+              if (FileUtils.directoryContains(inbox, dir)) {
+                /* cut away inbox path and trailing slash from artifact path */
+                seriesID = dir.getName();
+                if (seriesService.getSeries(seriesID) != null) {
+                  logger.info("Ingest from inbox into series with id {}", seriesID);
+                } else {
+                  seriesID = null;
+                }
+              }
+
+              /* Add title */
+              DublinCoreCatalog dcc = DublinCores.mkOpencastEpisode().getCatalog();
+              dcc.add(DublinCore.PROPERTY_TITLE, artifact.getName());
+              if (StringUtils.isNotBlank(seriesID))
+                dcc.add(DublinCore.PROPERTY_IS_PART_OF, seriesID);
+              logger.debug("episode dublincore for the inbox file {}: {}", artifact.getName(), dcc.toXml());
```

Comment:
You always serialize this as XML regardless of the log level. you can check if the log level is debug using the logger.

Reply:
I implemented this change